python/servo/testing_commands.py

```diff
@@ -319,6 +319,7 @@ def test_tidy(self, all_files, no_progress, self_test, stylo):
         if self_test:
             return test_tidy.do_tests()
         else:
+            run_update(self.context.topdir, check_clean=True)
```

Comment:
I found the first diff in `.travis.yml`, we will run `./mach test-tidy --no-progress --self-test` in Travis.
With the diff in this line, we won't check `manifest_changed` under `--self-test`.
So, if we removed `bash etc/ci/manifest_changed.sh` in it, then we won't check manifest changed.

Besides, it might be better to check `manifest_changed` when there's any WPT related file changes?

Reply:
`--self-test` is for unit test. we don't need to update manifest for unit test.
I think the best way to detect WPT change is using `manifest_changed`, and this doesn't take much time, so this might be an Okay solution.

Reply:
Have you tested what happens now when you run `./mach test-tidy` when the manifest does need updating? What does the output look like? I'm concerned that we suppress any non-zero return value from run_update right now.

Reply:
@jdm you are right... error doesn't return non-zoro.
```log
tigercosmos@ubuntu:~/servo$ ./mach test-tidy
 0:11.98 INFO Diffing old and new manifests /home/tigercosmos/servo/tests/wpt/mozilla/meta/MANIFEST.json
 0:12.47 INFO Diffing old and new manifests /home/tigercosmos/servo/tests/wpt/metadata/MANIFEST.json
 0:14.73 /home/tigercosmos/servo/tests/wpt/metadata/MANIFEST.json  0  error  core-aam/erer.html in source but not in manifest.  (wpt-manifest)
 0:14.99 ERROR Manifest /home/tigercosmos/servo/tests/wpt/metadata/MANIFEST.json is outdated, use |./mach update-manifest| to fix.
Checking the config file...
Checking the wpt manifest file...
Checking directories for correct file extensions...
Checking files for tidiness...
  Progress: 100% (3/3)
tidy reported no errors.
```

Reply:
I have added system exit for this.

Reply:
In my personal opinion, I'll prefer to keep `run_update` returning its status and check the status to exit the program here.

```py
if self_test:
    return test_tidy.do_tests()
else:
    if run_update(self.context.topdir, check_clean=True):
        sys.exit(1) # Maybe we can `return 1` instead?
    return tidy.scan(not all_files, not no_progress, stylo=stylo) 

# OR

if self_test:
    return test_tidy.do_tests()
else:
    return run_update(self.context.topdir, check_clean=True) and tidy.scan(not all_files, not no_progress, stylo=stylo) 
```

Reply:
In my personal opinion, I'll prefer to keep `run_update` returning its status and check the status to exit the program here.

```py
if self_test:
    return test_tidy.do_tests()
else:
    if run_update(self.context.topdir, check_clean=True):
        sys.exit(1) # Maybe we can `return 1` instead?
    return tidy.scan(not all_files, not no_progress, stylo=stylo) 

# OR

if self_test:
    return test_tidy.do_tests()
else:
    return run_update(self.context.topdir, check_clean=True) and tidy.scan(not all_files, not no_progress, stylo=stylo) 
```

Reply:
sounds good to me